apps/prairielearn/elements/pl-external-grader-results/pl-external-grader-results.py

```diff
@@ -39,18 +43,18 @@ def ansi_to_html(output):
         return f"[Error converting ANSI to HTML: {e}]\n\n{output}"
 
 
-def prepare(element_html, data):
+def prepare(element_html: str, data: pl.QuestionData) -> None:
     element = lxml.html.fragment_fromstring(element_html)
-    required_attribs = []
-    optional_attribs = []
+    required_attribs: List[str] = []
+    optional_attribs: List[str] = []
     pl.check_attribs(element, required_attribs, optional_attribs)
 
 
-def round_value(val, digits=2):
+def round_value(val: float, digits: int = 2) -> str:
     return format(val, f".{digits}f").rstrip("0").rstrip(".")
 
 
-def render(element_html, data):
+def render(element_html: str, data: pl.QuestionData) -> str:
     if data["panel"] == "submission":
         html_params = {"submission": True, "graded": True, "uuid": pl.get_uuid()}
```

Comment:
A bunch of the lines below add keys to this dict, but these can be done inline (like the `"graded"` key, for example).

Reply:
Great call-out, I can clean-up this method quite a bit. (This would be so much nicer in TypeScript with the spread operator / being able to write `{ value }` instead of `{ "value": value }`; I'll see if there's something similar for Python.)